src/analyzer/protocol/quic/QUIC.spicy

```diff
@@ -22,19 +22,81 @@ public function decrypt_crypto_payload(
 ##############
 
 # Can we decrypt?
-function can_decrypt(long_header: LongHeaderPacket, context: ConnectionIDInfo, is_client: bool): bool {
+function can_decrypt(long_header: LongHeaderPacket, context: ConnectionIDInfo&, state: CryptoState&): bool {
 
   if ( ! long_header.is_initial )
     return False;
 
-  if ( is_client )
-    return ! context.client_initial_processed;
+  # Can only decrypt the responder if we've seen the initial destination conn id.
+  if ( ! state.is_orig && |context.initial_destination_conn_id| == 0 )
+    return False;
+
+  # Only attempt decryption if we haven't flushed some SSL data yet.
+  return ! state.flushed;
+}
+
+# Buffer all CRYPTO frames, which might be fragmented and unordered
+# across multiple INITIAL packets into this CyptoBuffer unit.
```

Comment:
Do I understand correctly that this could buffer potentially unbounded data (well, at most uint24 bytes)? I am not sure I understand why we need to do this, and would much prefer an approach which passes data on as soon as we see it. Doing that would likely also simplify the tracking for `complete` state which currently happens in multiple places.

Reply:
Yeah, good call out! We can put a limit on the number of bytes we feed into the sink and just throw if that's exceeded, but also directly forward to the SSL analyzer from within the CryptoBuffer unit. I'll dabble with that!

I'm wondering if Spicy should provide a feature/limit for data that is pending in a `sink` and throw if it's exceeded.

Reply:
I've pushed a fixup that reworks this to avoid the buffering. Let me know what you think. Thanks!